runtime/lua/vim/filetype/detect.lua

```diff
@@ -6,15 +6,14 @@ local function count_lines(bufnr)
 end
 
 ---@private
-local function get_lines(bufnr, start_lnum, end_lnum)
-  end_lnum = end_lnum or start_lnum
-  return vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, end_lnum, false)
-end
+local function get_lines(bufnr, start_lnum, end_lnum, opts)
+  if not end_lnum and not opts then
```

Comment:
probably nobody would be silly enough to call `get_lines(0, 1, nil, opts)`, but just to be safe and cover all possibilities ;)
```suggestion
  if not end_lnum then
```